test/jasmine/tests/hover_label_test.js

```diff
@@ -0,0 +1,245 @@
+var d3 = require('d3');
+
+var Plotly = require('@src/index');
+var Fx = require('@src/plots/cartesian/graph_interact');
+var Lib = require('@src/lib');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+
+describe('hover info', function() {
+    'use strict';
+
+    var mock = require('@mocks/14.json'),
+        evt = {
+                clientX: mock.layout.width/ 2,
+                clientY: mock.layout.height / 2
+        };
+
+    afterEach(destroyGraphDiv);
+
+    describe('hover info', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+
+        beforeEach(function(done) {
+            Plotly.plot(createGraphDiv(), mockCopy.data, mockCopy.layout).then(done);
```

Comment:
Nop. That won't because we call Plotly.plot with different data each time. 

We could only force update hovermode in gd directly without a plot call and test the labels then.  But that feels kind of dirty.